tests/rbd/rbd_flatten_image_feature_disable.py

```diff
@@ -0,0 +1,101 @@
+from ceph.parallel import parallel
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def flatten_image_feature_disable(rbd, pool_type, **kw):
+    """
+        Flatten the clone image while changing the image feature.
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+    """
+
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        snap = kw["config"][pool_type].get("snap", f"{pool}_{image}_snap")
+        clone = kw["config"][pool_type].get("clone", f"{pool}_{image}_clone")
+        snap_name = f"{pool}/{image}@{snap}"
+        feature_name = "fast-diff"
+
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        run_fio(image_name=image, pool_name=pool, client_node=client)
+
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Creation of snapshot {snap} failed")
+            return 1
+
+        if rbd.protect_snapshot(snap_name):
+            log.error(f"Snapshot protect failed for {snap_name}")
+            return 1
+
+        if rbd.create_clone(snap_name, pool, clone):
+            log.error(f"Creation of clone {clone} failed")
+            return 1
+
+        with parallel() as p:
+            p.spawn(rbd.disable_rbd_feature, pool, clone, feature_name)
+            p.spawn(rbd.flatten_clone, pool, clone)
+
+            # Verify the feature is disabled and not present in the image info
+            image_load = rbd.image_info(pool, clone)
+            if feature_name not in image_load.get("features", {}):
```

Comment:
You are checking **not in** condition but you say that feature is present in rbd info? Isn't it contradicting?

Reply:
Done - Have made the changes.